File: backend/src/main/java/com/study4you/course/service/LessonService.java
package com.study4you.course.service;

import com.study4you.course.dto.LessonRequest;
import com.study4you.course.dto.LessonResponse;
import com.study4you.course.dto.SectionRequest;
import com.study4you.course.dto.SectionResponse;

import java.util.List;
import java.util.UUID;

public interface LessonService {

    SectionResponse createSection(UUID courseId, SectionRequest request);

    List<SectionResponse> getSectionsByCourse(UUID courseId);

    LessonResponse createLesson(UUID courseId, LessonRequest request);

    LessonResponse updateLesson(UUID lessonId, LessonRequest request);

    void deleteLesson(UUID lessonId);

    List<LessonResponse> getLessonsByCourse(UUID courseId, UUID currentUserId);

    /** Full lesson content (video url) — only for enrolled users or admin */
    LessonResponse getLessonById(UUID lessonId, UUID currentUserId);
}

File: frontend/src/services/lessonService.ts
import apiClient from "./apiClient";
import type { LessonResponse, SectionResponse } from "./courseService";

export interface LessonRequest {
  title: string;
  description?: string;
  videoUrl?: string;
  duration?: number;
  orderIndex: number;
  isPreview?: boolean;
  sectionId?: string;
}

export interface SectionRequest {
  title: string;
  orderIndex: number;
}

const lessonService = {
  getLessonsByCourse: async (courseId: string): Promise<LessonResponse[]> => {
    const res = await apiClient.get(`/courses/${courseId}/lessons`);
    return res.data.data;
  },

  getLessonById: async (lessonId: string): Promise<LessonResponse> => {
    const res = await apiClient.get(`/lessons/${lessonId}`);
    return res.data.data;
  },

  createLesson: async (courseId: string, data: LessonRequest): Promise<LessonResponse> => {
    const res = await apiClient.post(`/courses/${courseId}/lessons`, data);
    return res.data.data;
  },

  updateLesson: async (lessonId: string, data: LessonRequest): Promise<LessonResponse> => {
    const res = await apiClient.put(`/lessons/${lessonId}`, data);
    return res.data.data;
  },

  deleteLesson: async (lessonId: string): Promise<void> => {
    await apiClient.delete(`/lessons/${lessonId}`);
  },

  getSections: async (courseId: string): Promise<SectionResponse[]> => {
    const res = await apiClient.get(`/courses/${courseId}/sections`);
    return res.data.data;
  },

  createSection: async (courseId: string, data: SectionRequest): Promise<SectionResponse> => {
    const res = await apiClient.post(`/courses/${courseId}/sections`, data);
    return res.data.data;
  },
};

export default lessonService;
